jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/params/GeneralParams.scala

```diff
@@ -147,3 +169,47 @@ trait HasBaseMarginCol extends Params {
   /** @group getParam */
   final def getBaseMarginCol: String = $(baseMarginCol)
 }
+
+trait HasNumClass extends Params {
+
+  /**
+   * number of classes
+   */
+  final val numClass = new IntParam(this, "numClass", "number of classes")
+
+  /** @group getParam */
+  final def getNumClass: Int = $(numClass)
+}
+
+private[spark] trait ParamMapFuncs extends Params {
+
+  def XGBoostToMLlibParams(xgboostParams: Map[String, Any]): Unit = {
+    for ((paramName, paramValue) <- xgboostParams) {
+      val name = CaseFormat.LOWER_UNDERSCORE.to(CaseFormat.LOWER_CAMEL, paramName)
```

Comment:
did I miss anything here? why it can exactly match the camel case in this PR to the underscore format in https://github.com/dmlc/xgboost/blob/master/doc/parameter.md?